ReactApp/src/misc/api-fetch-setup.js

```diff
@@ -110,12 +110,24 @@ function corsMiddleware(options, next) {
 	return next(options);
 }
 
-function createHeadersMiddleware(authToken) {
+function createHeadersMiddleware(authToken, siteID) {
 	return (options, next) => {
 		options.headers = options.headers || {};
 
+		// Replace some namespaces to work with WordPress.com REST API.
+		if (
+			options.path &&
+			siteID.length &&
+			!options.path.includes(`/sites/${siteID}/`)
+		) {
+			options.path = options.path
+				.replace('/wp/v2/', `/wp/v2/sites/${siteID}/`)
+				.replace('/wpcom/v2/', `/wpcom/v2/sites/${siteID}/`)
+				.replace('/oembed/1.0/', `/oembed/1.0/sites/${siteID}/`);
+		}
```

Comment:
This logic appears unrelated to headers. I suggest we relocate it to a new middleware function — e.g., `createPathModifcationMiddleware` or similar. 

However, given our long-term hope of merging this into the Gutenberg project, this feels out of place in GutenbergKit. I wonder if we should hoist this middleware definition into the host app. WDYT? 

```swift
private func getEditorConfiguration() -> WKUserScript {
    // ...
    let jsCode = """
    window.GBKit = {
        siteAPIURL: '\(siteAPIURL)',
        authToken: '\(authToken)',
        siteID: '\(siteIDString)',
        modifyAPIPath
    };
    function modifyAPIPath(path) {
      if (
        !path ||
        !siteID.length ||
        path.includes(`/sites/${siteID}/`)
      ) {
        return path;
      }

      return path
        .replace('/wp/v2/', `/wp/v2/sites/${siteID}/`)
        .replace('/wpcom/v2/', `/wpcom/v2/sites/${siteID}/`)
        .replace('/oembed/1.0/', `/oembed/1.0/sites/${siteID}/`);
    }
    // ...
}
```

```js
apiFetch.use(pathModificationMiddleware); 

// ...

function pathModificationMiddleware(options, next) {
  const{ modifyAPIPath = () => {} } = getGBKit();
  options.path = modifyAPIPath(options.path);
  return next(options);
}
```

Reply:
> However, given our long-term hope of merging this into the Gutenberg project, this feels out of place in GutenbergKit. I wonder if we should hoist this middleware definition into the host app. WDYT?

Good idea, it makes sense to have this logic coming from the host app. I'll update it 👍

Reply:
Updated in https://github.com/wordpress-mobile/GutenbergKit/pull/15/commits/23f71cbc44c666e61a588ffb60b83f3d2ee4e573

I also moved the logic for the `Authorization` header to the native side.